taxcalc/tests/test_calculate.py

```diff
@@ -89,133 +69,130 @@ def test_sequence():
+    params1 = Parameters()
+    reform1 = {2013: {'_II_em': np.array([4000]), '_II_em_cpi': False}}
+    params1.implement_reform(reform1)
+    # create a Calculator object using this policy-reform
+    puf = Records(data=TAX_DTA, weights=WEIGHTS, start_year=2009)
+    calc2 = Calculator(params=params1, records=puf)
+    # check that Parameters object embedded in Calculator object is correct
     assert all(calc2.params._II_em == np.array([4000]))
 
 
-def test_make_Calculator_json():
-
-    # Create a Params object
-    params = Parameters(start_year=1991, inflation_rates=irates)
-
-    # Create a Public Use File object
-    puf = Records(tax_dta)
-
-    user_mods = """{"1991": { "_STD_Aged": [[1500, 1250, 1200, 1500, 1500, 1200 ]],
-                     "_STD_Aged_cpi": false}}"""
-
-    calc2 = calculator(params, puf, mods=user_mods, _II_em_cpi=False,
-                       _II_em=np.array([4000]))
+def test_make_Calculator_with_policy_reform2():
+    # create a Parameters object and apply a policy reform
+    params2 = Parameters()
+    reform2 = {2013: {'_II_em': np.array([4000]), '_II_em_cpi': False,
+                      '_STD_Aged': [[1600, 1300, 1300, 1600, 1600, 1300]],
+                      "_STD_Aged_cpi": False}}
+    params2.implement_reform(reform2)
+    # create a Calculator object using this policy-reform
+    puf = Records(data=TAX_DTA, weights=WEIGHTS, start_year=2009)
+    calc2 = Calculator(params=params2, records=puf)
+    # check that Parameters object embedded in Calculator object is correct
+    assert calc2.current_year == 2013
     assert calc2.params.II_em == 4000
     assert_array_equal(calc2.params._II_em, np.array([4000] * 12))
-    exp_STD_Aged = [[1500, 1250, 1200, 1500, 1500, 1200]] * 12
+    exp_STD_Aged = [[1600, 1300, 1300, 1600, 1600, 1300]] * 12
     assert_array_equal(calc2.params._STD_Aged, np.array(exp_STD_Aged))
-    assert_array_equal(calc2.params.STD_Aged, np.array([1500, 1250, 1200, 1500,
-                                                        1500, 1200]))
-
-
-def test_make_Calculator_user_mods_as_dict():
-
-    # Create a Params object
-    params = Parameters(start_year=1991, inflation_rates=irates)
-
-    # Create a Public Use File object
-    puf = Records(tax_dta)
-
-    user_mods = {1991: {"_STD_Aged": [[1400, 1200, 1000, 1000, 1000, 1000]]}}
-    user_mods[1991]['_II_em'] = [3925, 4000, 4100]
-    user_mods[1991]['_II_em_cpi'] = False
-    calc2 = calculator(params, puf, mods=user_mods)
-    assert calc2.params.II_em == 3925
-    exp_II_em = [3925, 4000] + [4100] * 10
-    assert_array_equal(calc2.params._II_em, np.array(exp_II_em))
     assert_array_equal(calc2.params.STD_Aged,
-                       np.array([1400, 1200, 1000, 1000, 1000, 1000]))
-
-
-def test_make_Calculator_with_reform_after_first_year():
-    fyr = 2008
-
-    # specify reform in in 2010
-    user_mods = {2010: {"_STD_Aged": [[1501, 1202, 1502, 1203, 1504, 1204]]}}
-    user_mods[2010]['_II_em'] = [5000, 6000]
```

Comment:
we need to maintain tests that test reforms specified in this way. Here, a value for parameter `_II_em` is specified for 2010 and 2011. We need to maintain tests like this because this is how we process data from the web application.